pyorbital/astronomy.py

```diff
@@ -39,55 +40,97 @@
 import datetime
 
 import numpy as np
+from typing import Tuple, Union
 
 from pyorbital import dt2np
 
-F = 1 / 298.257223563  # Earth flattening WGS-84
-A = 6378.137  # WGS84 Equatorial radius
+# Astronomical constants
+AU = 149597870700.0  # Astronomical unit in meters
+EARTH_ flattening = 1 / 298.257223563  # Earth flattening WGS-84
+EARTH_RADIUS = 6378.137  # WGS84 Equatorial radius in km
 MFACTOR = 7.292115E-5
+EARTH_SEMI_MAJOR_AXIS = 1.00000261 * AU  # Earth semi-major axis in meters
+EARTH_ECCENTRICITY = 0.01671123  # Earth orbital eccentricity
+EARTH_YEAR = 365.25636  # Length of Earth year in days
+PERIHELION_DAY = 3  # Day of year with Earth in perihelion (approx.)
 
+# Type alias for time arguments
+TimeLike = Union[dt.datetime, np.datetime64, np.ndarray]
 
-def jdays2000(utc_time):
+
+def jdays2000(utc_time: TimeLike) -> Union[float, np.ndarray]:
     """Get the days since year 2000.
+
+    :param utc_time: UTC time
+    :type utc_time: datetime.datetime, numpy.datetime64, or array-like
+    :return: Days since year 2000
+    :rtype: float or numpy.ndarray
     """
     return _days(dt2np(utc_time) - np.datetime64('2000-01-01T12:00'))
 
 
-def jdays(utc_time):
-    """Get the julian day of *utc_time*.
+def jdays(utc_time: TimeLike) -> Union[float, np.ndarray]:
+    """Get the Julian day of *utc_time*.
+
+    :param utc_time: UTC time
+    :type utc_time: datetime.datetime, numpy.datetime64, or array-like
+    :return: Julian day
+    :rtype: float or numpy.ndarray
     """
     return jdays2000(utc_time) + 2451545.0
 
 
-def _days(dt):
+def _days(dt: TimeLike) -> Union[float, np.ndarray]:
     """Get the days (floating point) from *d_t*.
+
+    :param dt: Time delta
+    :type dt: datetime.timedelta or numpy.timedelta64
```

Comment:
Some of these type annotations and docstrings are incorrect I think. This function for example takes `datetime.timedelta` objects or a numpy array of `timedelta64` objects. You have `TimeLike` listed here.